src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,105 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_FACTOR = 2;
+    private int capacity = 16;
+    private Node<K, V>[] table = (Node<K, V>[]) new Node[capacity];
+    private int size;
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkSize();
+        int index = getIndex(key);
+        if (table[index] == null) {
+            addNewNode(key, value);
+            return;
+        }
+        Node<K, V> currentNode = table[index];
+        if (Objects.equals(key, currentNode.key)) {
+            changeValue(currentNode, value);
+            return;
+        }
+        while (!Objects.equals(key, currentNode.key)) {
+            if (currentNode.next == null) {
+                currentNode.next = new Node<>(key, value);
+                size++;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        changeValue(currentNode, value);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> currentNode = table[getIndex(key)];
+        if (currentNode == null) {
+            return null;
+        }
+        if (Objects.equals(currentNode.key, key)) {
+            return currentNode.value;
+        }
+        while (!Objects.equals(currentNode.key, key)) {
+            if (currentNode.next == null) {
+                return null;
+            }
+            currentNode = currentNode.next;
+        }
+        return currentNode.value;
```

Comment:
You can simplify this method by removing the two if condition and changing the condition of the loop and adding a condition in loop.
```
if (Objects.equals(currentNode.key, key)) {
            return currentNode.value;
        }
```